math/modular_exponentiation.cpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * @file
+ * @brief C++ Program for Modular Exponentiation Iteratively.
+ * Calculate the value of an integer a raised to an integer exponent b 
+ * under modulo c. 
+ * @note The time complexity of this approach is O(log b).
+ *
+ * Example:
+ * (4^3) % 5 (where ^ stands for exponentiation and % for modulo)
+ * (4*4*4) % 5
+ * (4 % 5) * ( (4*4) % 5 )
+ * 4 * (16 % 5)
+ * 4 * 1
+ * 4
+ * We can also verify the result as 4^3 is 64 and 64 modulo 5 is 4
+ */
+
+#include <iostream> 
+
+
+/// Iterative Function to calculate a raised to exponent b 
+/// under modulo c in O(log b) using modular exponentiation.
+int power(int a, unsigned int b, int c) 
```

Comment:
Function documentation is not up to the [repository standards](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md).
Please see the [typical structure](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) of a program, that might help. 🙂